src/intervals/interval.jl

```diff
@@ -0,0 +1,176 @@
+
+# Base interval types and utilities
+# ---------------------------------
+bitstype 8 Strand
+
+convert(::Type{Strand}, strand::Uint8) = box(Strand, unbox(Uint8, strand))
+convert(::Type{Uint8}, strand::Strand) = box(Uint8, unbox(Strand, strand))
+
+
+const STRAND_NA   = convert(Strand, 0b000)
+const STRAND_POS  = convert(Strand, 0b001)
+const STRAND_NEG  = convert(Strand, 0b010)
+const STRAND_BOTH = convert(Strand, 0b011)
+
+function Base.show(io::IO, strand::Strand)
+    if strand == STRAND_NA
+        print(io, "(indeterminate strand)")
```

Comment:
If we stick to the convention of GFF3, showing `"?"` instead of `"(indeterminate strand)"` would be more concise and not disturb alignment of texts.